deployment/windows/jellyfin.nsi

```diff
@@ -0,0 +1,252 @@
+    WriteRegStr HKLM "Software\Microsoft\Windows\CurrentVersion\Uninstall\Jellyfin" "DisplayIcon" '"$INSTDIR\Jellyfin.exe",0'
+    WriteRegStr HKLM "Software\Microsoft\Windows\CurrentVersion\Uninstall\Jellyfin" "Publisher" "The Jellyfin project"
+    WriteRegStr HKLM "Software\Microsoft\Windows\CurrentVersion\Uninstall\Jellyfin" "URLInfoAbout" "https://jellyfin.github.io/"
+    WriteRegStr HKLM "Software\Microsoft\Windows\CurrentVersion\Uninstall\Jellyfin" "DisplayVersion" "$_JELLYFINVERSION_"
+    WriteRegDWORD HKLM "Software\Microsoft\Windows\CurrentVersion\Uninstall\Jellyfin" "NoModify" 1
+    WriteRegDWORD HKLM "Software\Microsoft\Windows\CurrentVersion\Uninstall\Jellyfin" "NoRepair" 1
+
+;Create uninstaller
+    WriteUninstaller "$INSTDIR\Uninstall.exe"
+
+SectionEnd
+
+Section /o "Jellyfin desktop shortcut" DesktopShortcut
+    SetShellVarContext current ; to create the shortcut for current user only
+    DetailPrint "Creating desktop shortcut"
+    CreateShortCut "$DESKTOP\Jellyfin.lnk" "$INSTDIR\jellyfin.exe"
+SectionEnd
+
+Section "Jellyfin Service" InstallService
+    ExecWait '"$INSTDIR\nssm.exe" install Jellyfin "$INSTDIR\jellyfin.exe" --datadir "$_JELLYFINDATADIR_"' $0
+    DetailPrint "Jellyfin Service install, $0"
+    Sleep 3000
+    ExecWait '"$INSTDIR\nssm.exe" set Jellyfin Start SERVICE_DELAYED_AUTO_START' $0
+    DetailPrint "Jellyfin Service setting, $0"
+    StrCpy $_SERVICEINSTALLED_ "YES"
+SectionEnd
+
+Section "Start Jellyfin after installation" LaunchJellyfin
+; either start the service or launch jellyfin standalone
+    StrCmp $_SERVICEINSTALLED_ "YES" ServiceStart Standalone
+    
+    ServiceStart:
+    ExecWait '"$INSTDIR\nssm.exe" start Jellyfin' $0
+    DetailPrint "Jellyfin service start, $0"
+    Return
+    
+    Standalone:
+    ExecWait '"$INSTDIR\jellyfin.exe"' $0
+    DetailPrint "$INSTDIR\jellyfin.exe start, $0"
+	Return
+
+SectionEnd
+
+;--------------------------------
+;Descriptions
+
+;Language strings
+    LangString DESC_InstallJellyfin ${LANG_ENGLISH} "Install Jellyfin"
+    LangString DESC_InstallService ${LANG_ENGLISH} "Install As a Service"
+    LangString DESC_DesktopShortcut ${LANG_ENGLISH} "Create a desktop shortcut"
+    LangString DESC_LaunchJellyfin ${LANG_ENGLISH} "Start Jellyfin after Install"
+
+;Assign language strings to sections
+    !insertmacro MUI_FUNCTION_DESCRIPTION_BEGIN
+    !insertmacro MUI_DESCRIPTION_TEXT ${InstallJellyfin} $(DESC_InstallJellyfin)
+    !insertmacro MUI_DESCRIPTION_TEXT ${DesktopShortcut} $(DESC_DesktopShortcut)
+    !insertmacro MUI_DESCRIPTION_TEXT ${InstallService} $(DESC_InstallService)
+    !insertmacro MUI_DESCRIPTION_TEXT ${LaunchJellyfin} $(DESC_LaunchJellyfin)
+    !insertmacro MUI_FUNCTION_DESCRIPTION_END
+
+;--------------------------------
+;Uninstaller Section
+
+Section "Uninstall"
+    SetShellVarContext current
+    StrCpy $_JELLYFINDATADIR_ "$LOCALAPPDATA\jellyfin\"
```

Comment:
We should read this from the server config. (Although maybe that is something for later. We might just add a `jellyfin --full-reset` command to just wipe everything (transcode temp and all).